docs/raw/organisation-channels.md

```diff
@@ -0,0 +1,160 @@
+  string ens_name = 1;
+  string profile_picture = 2; // not a string, more complex data structure, but you get the point
+  string display_name = 3; 
+  string description = 4; // this is actually unique to an organisation, and chatMessageIdentity might want to be updated, or similar
+
+}
+message OrganisationPermissions {
+  enum Access {
+    UNKNOWN_ACCESS = 0;
+    NO_MEMBERSHIP = 1;
+    INVITATION_ONLY = 2;
+    ON_REQUEST = 3;
+  }
+
+  bool ens_only = 1;
+  bool can_request_access = 2;
+  // https://gitlab.matrix.org/matrix-org/olm/blob/master/docs/megolm.md is a candidate for the algorithm to be used in case we want to have private organisational chats, lighter than pairwise encryption using the DR, less secure, but more efficient for large number of participants
+  bool private = 3;
+  Access access = 4;
+}
+
+message OrganisationDescription {
+  uint64 clock = 1;
+  repeated bytes members = 2;
+  OrganisationPermissions permissions = 3;
+  ChatMessageIdentity identity = 5;
+  repeated OrganisationChat chats = 6;
+}
+
+message OrganisationChat {
+  string chat_id = 1;
+  uint64 clock = 2;
+  repeated bytes members = 3;
+  OrganisationPermissions permissions = 4;
+  string name = 5;
+}
+```
+
+Where:
+
+`clock` is a logical clock, described [here](link to clock value)
+`members` is a list of compressed public keys that represent the members of the organisation
+`identity` is a `ChatMessageIdentity` with a list of details about the organisation identity
+`permissions` is a list of permissions set by the creator of the organisation
+`chats` is a list of chats/(channels?) associated with the organisation
+
+
+### Permissions
+
+Permissions can be set at the organisation level or at the chat level.
+
+Permissions at the chat level cannot override permissions at the organisation level.
+(Here I will describe rules, but most of them should be simple, i.e you can't set a
+channel public in an private organisation, you can't have a member of a chat who is not
+a member of an organisation)
+
+`ens_only` means that only user who have an `ENS` can join the organisation
+`private` indicates whether the organisation channels are encrypted with a secret
+key shared only by members, otherwise the key is derivable by the chat_id
+`access` indicates how membership works for this organisation. 
+When is `NO_MEMBERSHIP` anyone can read and write on the channels, `private` MUST be set to false in this case.
+When is `INVITE_ONLY` users MUST NOT request access, if `private` is set to `false` the channel
+will be read-only. Only upon receiving an `OrganisationInvitation` a user will be able to join.
+When is `ON_REQUEST` users MUST send a `OrganisationRequestJoin` to join the organisation.
+
+
+## Joining an organisation
+
+If an organisation access is set to `ON_REQUEST` a user MUST send a
+message `OrganisationRequestJoin` if they want to join the organisation.
+A `chat_id` MAY also be specified if they are interested in joining a particular channel
```

Comment:
is the association of `chat_id`s and subscribed members stored in the `members` field of `OrganisationDescription`? if not, I think it needs to be stored somewhere? this info would be necessary to enforce per channel access control